jvm/ml4ir-inference/src/main/java/ml4ir/inference/tensorflow/utils/ModelFeatures.java

```diff
@@ -0,0 +1,158 @@
+package ml4ir.inference.tensorflow.utils;
+
+import com.google.common.collect.ImmutableMap;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ModelFeatures {
+    private QueryKey query_key;
+    private Rank rank;
+    private Label label;
+    private List<InputFeatures> features;
+
+    public ModelFeatures() {}
+
+    public QueryKey getQuery_key() {
+        return query_key;
+    }
+
+    public Rank getRank() {
+        return rank;
+    }
+
+    public Label getLabel() {
+        return label;
+    }
+
+    public List<InputFeatures> getFeatures() {
+        return features;
+    }
+
+    public Map<String, Map<String, String>> getDataTypesForFeatures() {
+        Map<String, Map<String, String>> data = new HashMap<>();
+        data.put(this.getLabel().getTfrecord_type(),
+                ImmutableMap.of(this.getLabel().getName(), this.getLabel().getDtype()));
+        data.put(this.getRank().getTfrecord_type(),
+                ImmutableMap.of(this.getRank().getName(), this.getRank().getDtype()));
+        data.put(this.getQuery_key().getTfrecord_type(),
+                ImmutableMap.of(this.getQuery_key().getName(), this.getLabel().getDtype()));
+        features.forEach(f -> data.put(f.getTfrecord_type(),
+                ImmutableMap.of(f.getName(), f.getDtype())));
+        return data;
+    }
+
+    public  Map<String, Map<String, String>> getDefaultValuesForFeatures() {
+        Map<String, Map<String, String>> data = new HashMap<>();
```

Comment:
`return ImmutableMap.of(getLabel().getTfrecord_type(), ImmutableMap.of(getLabel().getName(), ...`

Reply:
(i.e. no need to new HashMap<>() and a bunch of data.put() calls.)

Reply:
The last line for this has features.forEach(f->data.put(...)). I am not sure if we can do this inside ImmutableMap.of()

Reply:
ah that's true, you'd need to do 
```
features.stream().map(f -> ImmutableMap.of(f.getTfrecord_type(), ImmutableMap.of(f.getName(), f.getDefault_value())).collect(...
```
and this last "..." would need to be a Collector which combines many `Map<String, Map<String, String>>` into one.  I imagine there are some out there, predefined?

Reply:
I updated it but still it is not the cleanest implementation. I am open to ideas to make this better